internal/controllers/machinedeployment/machinedeployment_rollout_rollingupdate_test.go

```diff
@@ -596,6 +691,446 @@ func Test_reconcileOldMachineSetsRollingUpdate(t *testing.T) {
 	}
 }
 
+func TestReconcileInPlaceUpdateIntent(t *testing.T) {
+	utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.InPlaceUpdates, true)
+
+	testCases := []struct {
+		name                   string
+		md                     *clusterv1.MachineDeployment
+		newMS                  *clusterv1.MachineSet
+		oldMS                  []*clusterv1.MachineSet
+		machines               []*clusterv1.Machine
+		scaleIntents           map[string]int32
+		upToDateResults        map[string]mdutil.UpToDateResult
+		canUpdateAnswer        map[string]bool
+		expectedCanUpdateCalls map[string]bool
+		expectMoveFromMS       []string
+		expectScaleIntents     map[string]int32
+	}{
+		// NewMS not scaling up
+
+		{
+			name:  "Move replicas from oldMS to newMS",
+			md:    createMD("v1", 2),
+			newMS: createMS("ms2", "v2", 1),
+			oldMS: []*clusterv1.MachineSet{
+				createMS("ms1", "v1", 1),
+			},
+			machines: []*clusterv1.Machine{
+				createM("m1", "ms1", "v1"),
+				createM("m2", "ms2", "v2"),
+			},
+			scaleIntents: map[string]int32{},
+			upToDateResults: map[string]mdutil.UpToDateResult{
+				"ms1": {EligibleForInPlaceUpdate: true},
+			},
+			expectedCanUpdateCalls: map[string]bool{
+				"ms1": true,
+			},
+			canUpdateAnswer: map[string]bool{
+				"ms1": true,
+			},
+			expectMoveFromMS:   []string{"ms1"},
+			expectScaleIntents: map[string]int32{},
+		},
+		{
+			name:  "Do not move replicas from oldMS to newMS when newMS already have all the desired replicas",
+			md:    createMD("v1", 3),
+			newMS: createMS("ms2", "v2", 3),
+			oldMS: []*clusterv1.MachineSet{
+				createMS("ms1", "v1", 1),
+			},
+			machines: []*clusterv1.Machine{
+				createM("m1", "ms1", "v1"),
+				createM("m2", "ms2", "v2"),
+				createM("m3", "ms2", "v2"),
+				createM("m4", "ms2", "v2"),
+			},
+			scaleIntents: map[string]int32{},
+			upToDateResults: map[string]mdutil.UpToDateResult{
+				"ms1": {EligibleForInPlaceUpdate: true},
+			},
+			expectedCanUpdateCalls: map[string]bool{},
+			canUpdateAnswer:        map[string]bool{},
+			expectMoveFromMS:       []string{},
+			expectScaleIntents:     map[string]int32{},
+		},
+		{
+			name:  "Do not move replicas from oldMS to newMS when oldMS doesn't have replicas anymore",
+			md:    createMD("v1", 1),
```

Comment:
This was previously set to 3.
I think with 1 we are now hitting a different code branch
```
	// If new MS already has all desired replicas, it does not make sense to perform more in-place updates.
	if ptr.Deref(p.newMS.Spec.Replicas, 0) >= ptr.Deref(p.md.Spec.Replicas, 0) {
		return nil
	}
```